### PR TITLE
Added AirCCPropertyListing.cs

### PR DESCRIPTION
**Pull request by CCianfloneDev**
Requesting DZ-FSDev to review code.

### CODE REVIEW

File: AirCAndC/AirCCPropertyListing.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+
+namespace AirCandC_api
+{
+    /// <summary>
+    /// Represents a rental property listing on Air c&c.
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Air C&C

File: AirCAndC/AirCCPropertyListing.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    /// </summary>
+    public class AirCCPropertyListing
+    {
+        private int numberOfRooms;
+        private int numberOfBathrooms;
+        private int squareFootage;
+        private string address;
+        private RenterPackages rentTypeOffered;
+        private int priceForPackage;
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Convert value-type fields to props.

File: AirCAndC/AirCCPropertyListing.cs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        private List<Amenities> amenities = new List<Amenities>();
+
+        /// <summary>
+        /// Initializes a listing on Air c&c
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Air C&C

File: AirCAndC/AirCCPropertyListing.cs [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        /// <returns>Amenities offered.</returns>
+        public List<Amenities> GetAmenitiesOffered()
+        {
+            return this.amenities;
+        }
+
+        /// <summary>
+        /// Prints amenities associated with property.
+        /// </summary>
+        public void PrintAmenities()
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Consider changing to `GetAmenitiesString() : string` after impl.

File: AirCAndC/AirCCPropertyListing.cs [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        /// Sets amenities offered on property.
+        /// </summary>
+        /// <param name="amenities">Amenities offered.</param>
+        public void SetAmenitiesOffered(List<Amenities> amenities)
+        {
+            this.amenities = amenities;
+        }
+
+        /// <summary>
+        /// Returns a string representation of the listing on Air c&c.
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Air C&C

File: AirCAndC/AirCCPropertyListing.cs [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Sets amenities offered on property.
+        /// </summary>
+        /// <param name="amenities">Amenities offered.</param>
+        public void SetAmenitiesOffered(List<Amenities> amenities)
+        {
+            this.amenities = amenities;
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Consider setting a clone of the list to the field to uphold Encapsulation.

File: AirCAndC/AirCCPropertyListing.cs [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+        /// Prints amenities associated with property.
+        /// </summary>
+        public void PrintAmenities()
+        {
+
+            foreach (Amenities amenities in this.amenities)
+            {
+                Console.WriteLine($" {amenities} ");
+            }
+            if (amenities.Count <= 0)
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Optimization: This should happen before the loop.

File: AirCAndC/AirCCPropertyListing.cs [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+            this.priceForPackage = priceForPackage;
+        }
+
+        /// <summary>
+        /// Gets list of amenities offered on property.
+        /// </summary>
+        /// <returns>Amenities offered.</returns>
+        public List<Amenities> GetAmenitiesOffered()
+        {
+            return this.amenities;
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Consider returning a clone of the list to uphold Encapsulation.

**Review by DZ-FSDev — CHANGES_REQUESTED** [review]
**Change Requests for AirCCPropertyListing.cs**

File: AirCAndC/AirCCPropertyListing.cs (hunk 8 above)
Comment (CCianfloneDev) [reply AirCAndC/AirCCPropertyListing.cs]: Changed method to return clone of list to uphold encapsulation, however this method is not needed at this time, therefor I have commented it out for the time being.

File: AirCAndC/AirCCPropertyListing.cs [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+        }
+
+        /// <summary>
+        /// Gets price for rent type offered.
+        /// </summary>
+        public int PriceForPackage
+        {
+            get;
+            private set;
+        }
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] Currency amount should be changed to use `decimal`

File: AirCAndC/AirCCPropertyListing.cs [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+        /// </summary>
+        /// <param name="numberOfRooms">Number of rooms.</param>
+        /// <param name="numberOfBathrooms">Number of bathrooms.</param>
+        /// <param name="squareFootage">Square footage of property.</param>
+        /// <param name="address">Address associated with property.</param>
+        /// <param name="rentTypeOffered">Rent package associated with this listing (daily, weekly, monthly)</param>
+        /// <param name="priceForPackage">Price associated with renting package chosen.</param>
+        /// <param name="amenities">Amenities associated with this listing.</param>
+        public AirCCPropertyListing(int numberOfRooms, int numberOfBathrooms, int squareFootage, 
+            string address, RenterPackages rentTypeOffered, int priceForPackage, List<Amenities> amenities)
```
Comment (DZ-FSDev) [inline AirCAndC/AirCCPropertyListing.cs]: - [x] `priceForPackage` should be changed to the `decimal` tyoe

**Review by DZ-FSDev — CHANGES_REQUESTED** [review]
**CR for #14**
Looks good! I missed this on my initial review:

**Review by DZ-FSDev — APPROVED** [review]
Looks good!

**Pull request merged**